pkg/operator/ceph/csi/spec.go

```diff
@@ -392,16 +395,35 @@ func startDrivers(clientset kubernetes.Interface, rookclientset rookclient.Inter
 		}
 	}
 
-	tp.ProvisionerReplicas = 2
+	tp.ProvisionerReplicas = defaultProvisionerReplicas
 	nodes, err := clientset.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
 	if err == nil {
 		if len(nodes.Items) == 1 {
 			tp.ProvisionerReplicas = 1
 		}
 	} else {
-		logger.Errorf("failed to get nodes. Defaulting the number of replicas of provisioner pods to 2. %v", err)
+		logger.Errorf("failed to get nodes. Defaulting the number of replicas of provisioner pods to %d. %v", defaultProvisionerReplicas, err)
 	}
 
+	// retrieve CSI_PROVISIONER_REPLICAS only if the node count is greater than
+	// 1
+	if nodes != nil && len(nodes.Items) > 1 {
+		replicas, err := k8sutil.GetOperatorSetting(clientset, controllerutil.OperatorSettingConfigMapName, "CSI_PROVISIONER_REPLICAS", "")
+		if err != nil {
+			// logging a warning and intentionally continuing with the default
+			// provisioner replicas
+			logger.Warningf("failed to load CSI_PROVISIONER_REPLICAS. Defaulting to %d. %v", defaultProvisionerReplicas, err)
+		}
```

Comment:
We don't want to parse `replicas` if retrieving the setting failed, right?
```suggestion
		} else {
```

Reply:
returning an error if we fail to fetch the configuration.